src/game/gun/gun_misc.c

```diff
@@ -241,14 +246,86 @@ void Gun_GetNewTarget(WEAPON_INFO *winfo)
             && ang[1] >= winfo->lock_angles[2]
             && ang[1] <= winfo->lock_angles[3]) {
             int16_t yrot = ABS(ang[0]);
+            m_TargetList[num_targets] = item;
+            num_targets++;
             if (yrot < bestyrot) {
                 bestyrot = yrot;
                 bestitem = item;
             }
         }
     }
+    m_TargetList[num_targets] = NULL;
+
+    if ((g_Config.target_lock_mode == TLM_FULL
+         || g_Config.target_lock_mode == TLM_SEMI)
+        && g_Input.action && g_Lara.target) {
+        Gun_TargetInfo(winfo);
+        return;
+    }
+
+    if (num_targets > 0) {
+        for (int slot = 0; slot < NUM_SLOTS; slot++) {
+            if (!m_TargetList[slot]) {
+                g_Lara.target = NULL;
+            }
+
```

Comment:
Can these four lines be replaced with `for (int slot = 0; slot < num_targets; slot++) {`?

Reply:
I think that should work.

Reply:
Update: reverted this back to the original TR4 code since this change broke targeting.